core/azure-core/src/main/java/com/azure/core/util/polling/PollResponse.java

```diff
@@ -59,18 +67,31 @@
      * {@codesnippet com.azure.core.util.polling.pollresponse.status.value.retryAfter.properties}
      *
      * @param status Mandatory operation status as defined in {@link OperationStatus}.
+     * @param otherStatus string representation of custom status. It must be not null and non empty for {@link OperationStatus#OTHER} status,
+     * {@code null} is valid value for all other {@link OperationStatus}.
      * @param value The value as a result of poll operation. This can be any custom user-defined object. Null is also valid.
      * @param retryAfter Represents the delay the service has requested until the next polling operation is performed.
      * A {@code null}, zero or negative value will be taken to mean that the {@link Poller} should determine on its own when the next poll operation is to occur.
      * @param properties A map of properties provided by the service that will be made available into the next poll operation.
      * @throws NullPointerException If {@code status} is {@code null}.
+     * @throws IllegalArgumentException if otherStatus is null or empty when status is {@link OperationStatus#OTHER}.
      */
-    public PollResponse(OperationStatus status, T value, Duration retryAfter, Map<Object, Object> properties) {
+    public PollResponse(OperationStatus status, String otherStatus, T value, Duration retryAfter, Map<Object, Object> properties) {
```

Comment:
I would simplify the constructor - if the user is supply an 'otherStatus', you don't need them to supply an operation status, you can automatically make that OTHER.